OutOfSchool/OutOfSchool.WebApi.Tests/Services/Database/SensitiveWorkshopsServiceDBTests.cs

```diff
@@ -0,0 +1,211 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using AutoMapper;
+using FluentAssertions;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NUnit.Framework;
+using OutOfSchool.BusinessLogic.Models;
+using OutOfSchool.BusinessLogic.Models.Workshops;
+using OutOfSchool.BusinessLogic.Services;
+using OutOfSchool.BusinessLogic.Services.AverageRatings;
+using OutOfSchool.BusinessLogic.Services.Images;
+using OutOfSchool.BusinessLogic.Services.Workshops;
+using OutOfSchool.BusinessLogic.Util;
+using OutOfSchool.BusinessLogic.Util.Mapping;
+using OutOfSchool.Services;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Models.ChatWorkshop;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.Tests.Common;
+using OutOfSchool.Tests.Common.TestDataGenerators;
+
+namespace OutOfSchool.WebApi.Tests.Services.Database;
+
+[TestFixture]
+public class SensitiveWorkshopsServiceDBTests
+{
+    private DbContextOptions<OutOfSchoolDbContext> dbContextOptions;
+    private OutOfSchoolDbContext dbContext;
+
+    private ISensitiveWorkshopsService sensitiveWorkshopService;
+    private IWorkshopRepository workshopRepository;
+    private Mock<IEntityRepositorySoftDeleted<long, DateTimeRange>> dateTimeRangeRepository;
+    private Mock<IEntityRepositorySoftDeleted<Guid, ChatRoomWorkshop>> roomRepository;
+    private Mock<ITeacherService> teacherService;
+    private Mock<ILogger<WorkshopService>> logger;
+    private IMapper mapper;
+    private Mock<IImageDependentEntityImagesInteractionService<Workshop>> workshopImagesMediator;
+    private Mock<IProviderAdminRepository> providerAdminRepository;
+    private Mock<IAverageRatingService> averageRatingServiceMock;
+    private Mock<IProviderRepository> providerRepositoryMock;
+    private Mock<ICurrentUserService> currentUserServiceMock;
+    private Mock<IMinistryAdminService> ministryAdminServiceMock;
+    private Mock<IRegionAdminService> regionAdminServiceMock;
+    private Mock<ICodeficatorService> codeficatorServiceMock;
+    private Guid institutionId;
```

Comment:
(Unused fields) Hadn't checked all of them, but at least `logger` and `providerAdminRepository` are used only locally in `SetUp()`. Please consider moving them there as local variables (or probably even inline them). Please check other variables.

Reasoning: less class fields means you have less code, control less changes, know what is really needed for tests and what is needed only for instantiating `WorkshopService`